### PR TITLE
Create a new example that will run full ADF package (not from notebook)

### PR DESCRIPTION
**Pull request by mnlevy1981**
This example can be generalized to other outside diagnostic packages as well (e.g. ILAMB). The goal is to have the JupyterBook-generated web site link to the ADF website from the side bar.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributor's Guide](https://github.com/NCAR/CUPiD/wiki/Contributor's-Guide) (including the `pre-commit` check)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change? Other pull requests that create the same changes do not exist; however, this is dependent upon #142 

### New Feature Submissions:

1. [x] Does your submission pass tests?
2. [x] Have you lint your code locally prior to submission?

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you successfully tested your changes locally?

### CODE REVIEW

**Comment by TeaganKing** [comment]
The last thing we have to do here is to add a flag if ADF exists in build.py

File: cupid/build.py [hunk 1]
```diff
@@ -45,6 +46,14 @@ def build(config_path):
     subprocess.run(
         ["jupyter-book", "build", f"{run_dir}/computed_notebooks/{sname}", "--all"],
     )
+    adf_run = control["compute_notebooks"]["atm"]["link_to_ADF"]["parameter_groups"][
+        "none"
+    ]["adf_run"]
```
Comment (TeaganKing) [inline cupid/build.py:51]: Could generalize for all components. Also use .get() to set default `run_external_pkg` value.

**Comment by TeaganKing** [comment]
We should also add documentation on how to run ADF until #105 is resolved

File: examples/external_diag_packages/config.yml [hunk 2]
```diff
@@ -121,8 +121,10 @@ compute_notebooks:
             adf_root: ../../external_diag_packages/ADF/
             case_year_range: "1_100"
             base_case_year_range: "1995_2006"
-            adf_run: True
-            adf_vars: ['PRECT', 'SST']
+        external_tool:
+          tool_name: 'ADF'
```
Comment (TeaganKing) [inline examples/external_diag_packages/config.yml:125]: This is great! One thought-- if we eventually want to loop through multiple tools, should the ADF-specific parameters be a subset of the ADF tool?

**Comment by TeaganKing** [comment]
We may want to remove the dependency on CESM (rather than just CUPiD and ADF)-- see note in #142.

**Comment by mnlevy1981** [comment]
This runs for me with

```
$ cd ${CESM_ROOT}/tools/CUPiD/helper_scripts
$ ./generate_adf_config_file.py --cesm-root ${CESM_ROOT} --cupid-example external_diag_packages --adf-template ../externals/ADF/config_amwg_default_plots.yaml --out-file adf_config.yml
$ conda activate cupid-analysis
$ ../externals/ADF/run_adf_diag adf_config.yml # TOOK 75 seconds
$ cd ../examples/external_diag_packages/
$ conda activate cupid-dev
$ cupid-run
$ cupid-build
```

(assumes `cupid-dev` was points to ${CESM_ROOT}/tools/CUPiD/cupid`)

**Review by TeaganKing — APPROVED** [review]
This looks great and runs properly! Here's a screenshot of the output with the link to ADF working smoothly.
Screen Shot 2024-11-19 at 10 10 33 PM

**Pull request merged**